src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.RoslynWorkspace/RazorProjectInfoSerializer.cs

```diff
@@ -150,21 +150,37 @@ private static void WriteToFile(string configurationFilePath, RazorProjectInfo p
         File.Move(tempFileInfo.FullName, configurationFilePath);
     }
 
-    private static ImmutableArray<DocumentSnapshotHandle> GetDocuments(Project project, string projectPath)
+    internal static ImmutableArray<DocumentSnapshotHandle> GetDocuments(Project project, string projectPath)
     {
         using var documents = new PooledArrayBuilder<DocumentSnapshotHandle>();
 
         var normalizedProjectPath = FilePathNormalizer.NormalizeDirectory(projectPath);
 
-        // We go through additional documents, because that's where the razor files will be
-        // We could alternatively go through the Documents and look for our virtual C# documents, that the dynamic file info
-        // would have added
-        foreach (var document in project.AdditionalDocuments)
+        if (project.AdditionalDocumentIds.Count > 0)
```

Comment:
Is it intended that we never walk through `project.Documents` if there are only non-Razor additional documents in the workspace?

Reply:
It is, but now that you mention it, its probably a bit much of an assumption. I originally wrote this to process both, and it was getting complicated de-duping so I pulled it out, but in retrospect probably went too far. I like your suggestion a lot.